src/expanduser.cpp

```diff
@@ -200,4 +194,26 @@ boost::filesystem::path ExpandUser(const std::string& path)
     return {ReplaceString(path, "~", home_dir)};
 }
 
+#else
+
+MIOPEN_DECLARE_ENV_VAR(TEMP)
+
+boost::filesystem::path ExpandUser(const std::string& path)
+{
+    const char* home_dir = GetStringEnv(HOME{});
+    if(home_dir == nullptr)
+    {
+        home_dir = GetStringEnv(TEMP{});
+        if(home_dir == nullptr)
+        {
+            MIOPEN_THROW(miopenStatusInternalError);
+        }
+    }
```

Comment:
[Recommendation] I suspect that existing `GetHomeDir()` can do this job. And it should be able to handle environment variables like `TEMP`, or `TMP` etc.

Or you specifically need MIOPEN_THROW? or maybe `boost::filesystem::temp_directory_path()` is not working properly?

Reply:
There are more ways to access the `HOME` directory on Windows, not just querying the `HOME` variable. I changed the PR to address that. It also accounts for the different formats to represent a variable on Windows, e.g., PowerShell, CMD.exe, etc.